webcompat/api/uploads.py

```diff
@@ -42,21 +99,23 @@ def upload():
 
     Returns a JSON string that contains the filename and url.
     '''
-    if 'image' in request.files:
-        try:
-            file = request.files['image']
-            file_ext = file.filename.split('.')[-1]
-            today = date.today()
-            filename = os.path.join(str(today.year),
-                                    str(today.month),
-                                    str(uuid4()) + '.' + file_ext)
-            images.save(file, name=filename)
-            data = {
-                'filename': filename,
-                'url': images.url(filename)
-            }
-            return (json.dumps(data), 201, {'content-type': JSON_MIME})
-        except UploadNotAllowed:
-            abort(415)
+    if 'image' in request.files and request.files['image'].filename:
+        imagedata = request.files['image']
+    elif 'screenshot' in request.form:
```

Comment:
`screenshot` comes from an add-on `postMessage` https://github.com/webcompat/webcompat.com/blob/e275f75f8a86f891215260434aaaf7d5872e1e8f/webcompat/static/js/lib/bugform.js#L89-L117, if you're curious.